### PR TITLE
Add utility to update salt for caches

### PR DESCRIPTION
**Pull request by khalidharun**
This PR allows for cached functions to use an updated salt, which affects the table_name and shard_names.  
1. New shards will be created by copying data from old_shards.
2. Add new indexes for the new shards.
3. Add new entries into the `table_shard_map`.

Note: There was a previous method that allowed for updating salts but it led to more complications.

### CODE REVIEW

**Comment by khalidharun** [comment]
@peterhurford @kirillseva Can you review this? Wrote a nice test for this also.

File: R/migrations.R [hunk 1]
```diff
@@ -1,50 +1,40 @@
...
-  ## fcn_call$table  # this would be the old table name
-  ## ```
-  ## And repeating the same step for the new function.
-  ## then the inputs to this migrator should be the cached function itself
-  ## (for now only used for getting the right db connection)
-  ## and a list in the form of
-  ## `list("old_table_name_1" = "new_table_name_1", ..., "old_table_name_n" = "new_table_name_n")`
+update_cache_salt <- function(dbconn, prefix, old_salt, new_salt) {
+  ## Updating the salts of an existing cached function can be tricky. In order to do so,
+  ## the old shards should be migrated over to ensure while still preserving the old shards
```
Comment (kirillseva) [inline R/migrations.R]: could you refactor english? :)

File: R/migrations.R [hunk 2]
```diff
@@ -1,50 +1,40 @@
...
+  ## in case a single user doesn't update cachemeifyoucan in all their machines.
+
+  ## In this approach, we are simple copying over old data into new shards and adding new entries
```
Comment (kirillseva) [inline R/migrations.R]: typo

**Comment by kirillseva** [comment]
Code looks good!

**Comment by robertzk** [comment]
@khalidharun Nice! LGTM.

File: R/migrations.R (hunk 1 above)
Comment (khalidharun) [inline R/migrations.R]: haha.. I think I gave up on english in this description.

File: R/db.R [hunk 3]
```diff
@@ -72,7 +77,7 @@ db2df <- function(df, dbconn, key) {
 #' @param df. Raw fetched database table.
 #' @param dbconn SQLConnection. A database connection.
 #' @param key. Identifier of database table.
-add_index <- function(dbconn, tblname, key, idx_name) {
+add_index <- function(dbconn, tblname, key, idx_name = paste0("idx_", digest::digest(tblname))) {
```
Comment (peterhurford) [inline R/db.R:80]: Nice :+1:

File: tests/testthat/test-migrations.R [hunk 4]
```diff
@@ -0,0 +1,46 @@
+context('cache migrations')
+
+library(DBI)
+library(dbtest)
+library(testthatsomemore)
+
+describe("update_cache_salt", {
```
Comment (peterhurford) [inline tests/testthat/test-migrations.R:7]: :yellow_heart: tests!

**Comment by peterhurford** [comment]
Looks good.  Obviously be careful about how you use this. ;)

**Comment by khalidharun** [comment]
@robertzk This is good to merge! (yay)

File: tests/testthat/test-migrations.R (hunk 4 above)
Comment (robertzk) [inline tests/testthat/test-migrations.R:7]: Awesome!

**Comment by robertzk** [comment]
Love it!

**Comment by kirillseva** [comment]
shipit!

File: DESCRIPTION [hunk 5]
```diff
@@ -1,7 +1,7 @@
 Package: cachemeifyoucan
 Type: Package
 Title: Cache Me If You Can
-Version: 0.2.3.1
+Version: 0.2.3.2
```
Comment (peterhurford) [inline DESCRIPTION:4]: BTW Don't forget to update NEWS.md when you update DESCRIPTION

**Pull request merged**